### PR TITLE
Support for ResourceGroup Custom Assertion Using Id

### PR DESCRIPTION
**Pull request by ozboms**
- Added example tests for storage account using custom assertions.
- Added support for getting resource group information using `Id` property. This adds support for Synapse resources.

### CODE REVIEW

**Review by marclerwick — CHANGES_REQUESTED** [review]

File: Modules/BenchPress.Azure/Public/CustomAssertions/BeInResourceGroup.ps1 [hunk 1]
```diff
@@ -27,15 +27,22 @@ function ShouldBeInResourceGroup ($ActualValue, [string]$ExpectedValue, [switch]
...
   } else {
     if ([bool]$ActualValue.ResourceDetails.PSObject.Properties[$rgProperty]){
-      $resourceGroupName = $ActualValue.ResourceDetails.ResourceGroup
+      $resourceGroupName = $ActualValue.ResourceDetails.$rgProperty
     } elseif ([bool]$ActualValue.ResourceDetails.PSObject.Properties[$rgNameProperty]){
-      $resourceGroupName = $ActualValue.ResourceDetails.ResourceGroupName
+      $resourceGroupName = $ActualValue.ResourceDetails.$rgNameProperty
+    } elseif ([bool]$ActualValue.ResourceDetails.PSObject.Properties[$idProperty]){
+      # If it does not have a property for resource group but it has an Id,
+      # then we can get it from Id
```
[the remark this answers is not in the document]
Comment (ozboms) [reply Modules/BenchPress.Azure/Public/CustomAssertions/BeInResourceGroup.ps1:42]: I'll add a small sample comment. There is already a sample in the `BeInResourceGroup.Tests.ps1` on line 51:

`Id = '/subscriptions/<subscription-id>/resourceGroups/fakerg/providers/Microsoft.Synapse/workspaces/fakerg'`

And to your point about this expected format: it sadly isn't 100% the same. I tried to run this on the Key Vault Key/Certificate/Secret (the only ones that were not working before) and it threw an empty string error. I checked and the KeyVault Key does [have an Id field](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.commands.keyvault.models.pskeyvaultkey?view=az-ps-latest) but it's formatted differently/doesn't have the same information.

This is just another way for getting the information.

[the remark this answers is not in the document]
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/CustomAssertions/BeInResourceGroup.ps1:42]: Thanks, yeah, I saw the test before I wrote the comment. And although tests are a good way to understand some logic, it's much easier to have the explanation where the code executes from a maintenance perspective.

What is the behavior of the Custom Assertion when the empty string error is thrown and not caught? Does the rest of the existing logic continue or is it a terminating error? Is the empty string error displayed to the user? If it is a terminating error or if the empty string error is displayed to the user, then it should be caught and suppressed, and the existing logic will send the proper message.

[the remark this answers is not in the document]
Comment (ozboms) [reply Modules/BenchPress.Azure/Public/CustomAssertions/BeInResourceGroup.ps1:42]: It continues and fails at the If-statement below. It fails and shows the failure message.

` else {
              $failureMessage = "Resource not in resource group or there was an error when confirming resource.
              Expected $ExpectedValue but got $resourceGroupName."`

[the remark this answers is not in the document]
Comment (ozboms) [reply Modules/BenchPress.Azure/Public/CustomAssertions/BeInResourceGroup.ps1:42]: Added it + unit tests to match/check

**Review by marclerwick — APPROVED** [review]

**Pull request merged**